blockly-ws-search/test/workspace_search_test.js

```diff
@@ -0,0 +1,306 @@
+      assert.equal(actualGroup.length, expectedGroup.length);
+      for (var block, i = 0; (block = allBlocks[i]); i++) {
+        if (expectedGroup.indexOf(block) !== -1) {
+          assert.equal(actualGroup.indexOf(block) !== -1, true,
+              'Expected block [' + block.type + '] to be in search results');
+          assert.equal(isBlockHighlighted(block), true,
+              'Expected block [' + block.type + '] to be highlighted.');
+        } else {
+          assert.equal(actualGroup.indexOf(block) !== -1, false,
+              'Unexpected block [' + block.type + '] in search results');
+          assert.equal(isBlockHighlighted(block), false,
+              'Unexpected block [' + block.type + '] found highlighted.');
+        }
+      }
+    }
+    setup(() => {
+      Blockly.defineBlocksWithJsonArray([
+        {
+          'type': 'test_block',
+          'message0': 'test block'
+        },
+        {
+          "type": "test_statement_block",
+          "message0": "%test %1",
+          "args0": [{
+            "type": "input_value",
+            "name": "INPUT0",
+            "check": "String"
+          }],
+          "message1": "%block %1",
+          "args1": [{
+            "type": "input_statement",
+            "name": "INPUT1"
+          }],
+          "previousStatement": null,
+          "nextStatement": null
+        },
+        {
+          "type": "test_text",
+          "message0": "%1",
+          "args0": [{
+            "type": "field_input",
+            "name": "NAME",
+            "text": "test string"
+          }],
+          "output": null
+        }
+      ]);
+      this.testBlock = this.workspace.newBlock('test_block');
+      this.testStatementBlock = this.workspace.newBlock('test_statement_block');
+      this.testStatementBlockWithInput =
+          this.workspace.newBlock('test_statement_block');
+      this.fieldWithOutputConnected =
+          this.workspace.newBlock('test_text');
+      this.testStatementBlockWithInput.inputList[0].connection
+          .connect(this.fieldWithOutputConnected.outputConnection);
+      this.testStatementBlockWithInputCollapsed =
+          this.workspace.newBlock('test_statement_block');
+      this.fieldWithOutputCollapsed =
+          this.workspace.newBlock('test_text');
+      this.testStatementBlockWithInputCollapsed.inputList[0].connection
+          .connect(this.fieldWithOutputCollapsed.outputConnection);
+      this.testStatementBlockWithInputCollapsed.setCollapsed(true);
+      this.fieldWithOutput = this.workspace.newBlock('test_text');
+
+      this.blocks = [
+          this.testBlock,
+          this.testStatementBlock,
+          this.testStatementBlockWithInput,
+          this.testStatementBlockWithInputCollapsed,
+          this.fieldWithOutputConnected,
+          this.fieldWithOutputCollapsed,
+          this.fieldWithOutput
+      ];
+
+      sinon.stub(this.workspace, "getAllBlocks").returns(Object.values(this.blocks));
```

Comment:
Does this get used?

Reply:
Yes, it is called in `geSearchPool_` as part of the search in `WorkspaceSearch`.